### DISCUSSION TITLE
Skywalking redission has not more detail command to show

### DISCUSSION THREAD

**Discussion by CodingMakeWordBetter** [apache/skywalking#12479] [Q&A]
When I use Skywalk to view links, I find that Redisson's granularity of Redis operations is only at the instruction level. Can there be more specific instructions? For example, as shown in the figure, there is only one get, but I would prefer to know what got, similar to a MySQL statement. Get key. Is this feature not developed yet or is my usage incorrect?

**Comment by CodingMakeWordBetter** [comment]
![06807F49-8FE9-4D04-842B-C2C90E6087A0](https://github.com/user-attachments/assets/a3ae3af0-40bc-4c96-9c6f-1dd209963677)

**Comment by wu-sheng** [comment]
You could simply search `redisson` in the config doc, https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/configurations/

- `plugin.redisson.trace_redis_parameters`
- `plugin.redisson.redis_parameter_max_length`

**Reply by CodingMakeWordBetter** [reply]
I followed the document to modify the plugin.redison.trace.redis_parameters in skywalking agent/config/agentconfig to plugin.redison.trace.redis_parameters=${SW-PLLUGIN-REDISSON_TRACE-REDIVNet: true}
or
plugin.redisson.trace_redis_parameters=true.
After restarting the application, it still didn't take effect. I added the path specified in my Java startup command - DSkywalking_comg=/home/admin/skywalking agent/config/agentconfig, but it didn't work

**Reply by wu-sheng** [reply]
I can't debug for you. But the codes and config are there.

- v3, https://github.com/apache/skywalking-java/blob/main/apm-sniffer/apm-sdk-plugin/jedis-plugins/jedis-2.x-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/jedis/v3/JedisPluginConfig.java
- v4, https://github.com/apache/skywalking-java/blob/main/apm-sniffer/apm-sdk-plugin/jedis-plugins/jedis-4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/jedis/v4/JedisPluginConfig.java

**Reply by CodingMakeWordBetter** [reply]
I tried to solve this problem and found that the issue occurred in class org.apache.skywalking.apm.plugin.redisson.v3.RedisConnectionMethodInterceptor line 132 of the screenshot because I returned different keys in all if code blocks and then packaged and uploaded them locally to return the argument type to Skywalk. I found that it was [B, which is a byte array. However, I found it inconvenient to troubleshoot this way. I tried printing the allArguments parameter in the log, but I did not see the log I printed on/skywalking agent/logs/skywalking api.rog. Is there any suggestion for troubleshooting this inconvenience?
![97C380F3-3022-488B-8D89-DCFA0D3668D8](https://github.com/user-attachments/assets/2c0894d4-4f45-4773-803a-2c650ea80131)

**Reply by CodingMakeWordBetter** [reply]
I tried to solve the problem and found that this way I could report key data to Skywalking, as shown in the picture.I want to fix this problem. but i found skywalking-java just a main branch. i use version is v9.1.0.show i request merge to main branch?how aboult v9.1.0?
![9A1E8F94-9192-4B15-BF3F-530B8180CDEA](https://github.com/user-attachments/assets/7ecce199-86e7-4091-a2ab-bcac827d0c1d)

**Reply by wu-sheng** [reply]
All fixes should be landed on the main branch. If get merged, release on next(9.4.0).

**Reply by CodingMakeWordBetter** [reply]
I have send merge request to fix this problem.https://github.com/apache/skywalking-java/pull/712
